src/app/file/gif_format.cpp

```diff
@@ -561,18 +561,31 @@ class GifDecoder {
     // sprite palette.
     int found = 0;
     if (m_frameNum > 0) {
-      for (int i=0; i<ncolors; ++i) {
-        if (!usedEntries[i])
-          continue;
+      ColorMapObject* globalCMap = m_gifFile->SColorMap;
+      ColorMapObject* localCMap = m_gifFile->Image.ColorMap;
+      if (globalCMap && !localCMap)
+        found = usedEntries.size();
+      else {
+        for (int i=0; i<ncolors; ++i) {
+          if (!usedEntries[i])
+            continue;
+
+          if (localCMap && i < localCMap->ColorCount &&
+              rgba(localCMap->Colors[i].Red,
+                   localCMap->Colors[i].Green,
+                   localCMap->Colors[i].Blue, 256) == palette->getEntry(i)) {
```

Comment:
Didn't test this yet, but seeing the code, it looks like this 256 should be 255.
```suggestion
                   localCMap->Colors[i].Blue, 255) == palette->getEntry(i)) {
```